R/three-legged-oauth.R

```diff
@@ -1,21 +1,35 @@
 
 #' Create a token for a user via the browser flow.
-#' @param scopes List of scopes required for the returned token.
-#' @param oauth_app `httr::oauth_app` to use for this token fetch. (Optional.)
+#' @param scopes A character vector of scopes to request.
+#' @param app An OAuth consumer application, created by [httr::oauth_app()].
+#' @param cache A logical value or a string. `TRUE` means to cache using the
+#'   default cache file, `.httr-oauth`. `FALSE` means don't cache, and `NA`
+#'   means to guess using some sensible heuristics. A string means use the
+#'   specified path as the cache file.
 #' @param ... Additional arguments (ignored)
 #' @export
-credentials_user_oauth2 <- function(scopes, oauth_app = NULL, ...) {
-
-  app <- httr::oauth_app(
-    "google",
-    key = "603366585132-nku3fbd298ma3925l12o2hq0cc1v8u11.apps.googleusercontent.com",
-    secret = "as_N12yfWLRL9RMz5nVpgCZt"
-  )
+#' @examples
+#' \dontrun{
+#' scope <- "https://www.googleapis.com/auth/drive"
+#' app <- httr::oauth_app(
+#'   appname = "my_awesome_app",
+#'   key = "keykeykeykeykeykey",
+#'   secret = "secretsecretsecret"
+#' )
+#'
+#' ## credentials_user_oauth2() is one of the functions token_fetch() will try
+#' token_fetch(scope, app = app)
+#' }
+credentials_user_oauth2 <- function(scopes,
+                                    app,
+                                    cache = getOption("httr_oauth_cache"),
+                                    ...) {
   token <- httr::oauth2.0_token(
     endpoint = httr::oauth_endpoints("google"),
     app = app,
     scope = scopes,
-    use_oob = FALSE
+    use_oob = FALSE,
```

Comment:
I think we can work on the assumption that if we ever use something other than `oauth2.0_token` it would still have to use the same args. I think passing `...` is fine.